pkg/cmd/checkendpoint/controller/status_updater.go

```diff
@@ -0,0 +1,75 @@
+package controller
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/cmd/checkendpoint/operatorcontrolplane/podnetworkconnectivitycheck/v1alpha1helpers"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"k8s.io/klog"
+)
+
+type StatusUpdater interface {
+	sync.Locker
+	Start(ctx context.Context)
+	Stop()
+	Add(updates ...v1alpha1helpers.UpdateStatusFunc)
+}
+
+func NewStatusUpdater(client v1alpha1helpers.PodNetworkConnectivityCheckClient, checkName string, recorder events.Recorder) StatusUpdater {
+	return &statusUpdater{
+		checkName: checkName,
+		client:    client,
+		recorder:  recorder,
+		stop:      make(chan interface{}),
+	}
+}
+
+type statusUpdater struct {
+	sync.Mutex
+	checkName string
+	updates   []v1alpha1helpers.UpdateStatusFunc
+	client    v1alpha1helpers.PodNetworkConnectivityCheckClient
+	recorder  events.Recorder
+	stop      chan interface{}
+}
+
+func (s *statusUpdater) Add(updates ...v1alpha1helpers.UpdateStatusFunc) {
+	s.Lock()
+	defer s.Unlock()
+	s.updates = append(s.updates, updates...)
+}
+
+func (s *statusUpdater) Start(ctx context.Context) {
+	go func() {
+		for {
+			select {
+			case <-s.stop:
+				return
+			case <-ctx.Done():
+				return
+			case <-time.After(1 * time.Second):
+				s.updateStatus(ctx)
+			}
+		}
+	}()
+}
+
+func (s *statusUpdater) Stop() {
+	close(s.stop)
+}
+
+func (s *statusUpdater) updateStatus(ctx context.Context) {
+	s.Lock()
+	defer s.Unlock()
+	if len(s.updates) > 0 {
+		_, _, err := v1alpha1helpers.UpdateStatus(ctx, s.client, s.checkName, s.updates...)
+		if err != nil {
+			klog.Warningf("Unable to update %s: %v", s.checkName, err)
+			s.recorder.Warningf("UpdateFailed", "Unable to update %s: %v", s.checkName, err)
```

Comment:
do we really need an event for this? At worst this triggers every second.

Reply:
Done. Removed.